crates/db-address-space/src/address_space.rs

```diff
@@ -590,28 +610,34 @@ impl AddressSpaceInternal {
         Ok(())
     }
 
-    /// Get numa node id from region.
+    /// Enumerate all regions in the address space.
     ///
     /// # Arguments
-    /// * `gpa` - guest physical address.
-    pub fn get_numa_node_id(&self, gpa: u64) -> Option<u32> {
+    /// * `cb` - the callback function to apply to each region.
+    pub fn walk_regions<F>(&self, mut cb: F) -> Result<(), AddressSpaceError>
+    where
+        F: FnMut(&Arc<AddressSpaceRegion>) -> Result<(), AddressSpaceError>,
+    {
         for reg in self.regions.iter() {
-            if gpa >= reg.base.0 && gpa < (reg.base.0 + reg.size) {
-                return reg.host_numa_node_id;
-            }
+            cb(reg)?;
         }
 
-        None
+        Ok(())
     }
 
-    /// Get address space boundary
-    pub fn get_boundary(&self) -> AddressSpaceBoundary {
-        self.boundary.clone()
+    /// Create a [GuestMemoryMmap] object from the address space object.
+    pub fn create_guest_memory() -> GuestMemoryMmap {
```

Comment:
Is this method necessary?

Reply:
It's unnecessary once we make hotplug mandatory.